servers/cromwell/jobs/controllers/capabilities_controller.py

```diff
@@ -10,26 +10,38 @@ def get_capabilities():
     Returns:
         CapabilitiesResponse: Response containing this backend's capabilities
     """
-    capabilities = CapabilitiesResponse(
-        display_fields=[
-            DisplayField(field='status', display='Status'),
-            DisplayField(field='submission', display='Submitted'),
-            DisplayField(
-                field='labels.cromwell-workflow-name',
-                display='Workflow Name'),
-            DisplayField(
-                field='labels.cromwell-workflow-id', display='Workflow ID'),
-            DisplayField(field='labels.comment', display='Comment')
+    # Set default capabilities
+    capabilities = {
+        "display_fields": {
+            "status": "Status",
+            "submission": "Submitted",
+            "labels.cromwell-workflow-id": "Workflow ID"
+        },
+        "common_labels": [
+            "cromwell-workflow-id"
         ],
-        common_labels=[
-            'cromwell-workflow-name', 'cromwell-workflow-id', 'comment'
-        ],
-        query_extensions=[])
-    if current_app.config['use_caas']:
-        capabilities.authentication = AuthenticationCapability(
-            is_required=True,
-            scopes=[
-                'https://www.googleapis.com/auth/userinfo.profile',
-                'https://www.googleapis.com/auth/userinfo.email'
-            ])
-    return capabilities
+        "query_extensions": []
+    }
+
+    capabilities_config = current_app.config['capabilities_config']
+    if capabilities_config:
+        capabilities.update(capabilities_config)
+
+    display_fields, common_labels, query_extensions = [], set([]), set([])
+
+    # Construct display_fields
+    for field, display in capabilities['display_fields'].items():
```

Comment:
This loop provides the flexibility to add any display_field to the UI, but sacrifices the ability to control the order of columns. In order to control the order of the columns, we have three ways to do it:

1. Make changes to the UI so users can modify the order of columns in their browsers.
2. Control part of the orders(status and submission) but keep the flexibility:
```python
display_fields = [capabilities['display_fields'].pop('status'), capabilities['display_fields'].pop('submission')]

for field, display in capabilities['display_fields'].items():
        display_fields.append(DisplayField(field=field, display=display))
```
3. Control all of the orders:
Change the structure of ` capabilities_config.json` to:
```json
{
      "display_fields": {
        "status": "Status",
        "submission": "Submitted",
        "labels": [
          {"cromwell-workflow-id": "Workflow ID"},
          {"workflow-name": "Workflow Name"},
          {"comment": "Comment"}
        ]
      },
      "common_labels": [
        "cromwell-workflow-id",
        "workflow-name",
        "comment"
      ],
      "query_extensions": []
    }
```
and modify the `flatten` function(much more expensive) to:
```python
def flatten(dic, nest_keys, sep='.'):
    nest_keys = set(nest_keys)
    res_dic ={}
    for key, val in dic.items():
        if key in nest_keys and isinstance(val, list):
            res_dic[key] = []
            for inner_dic in val:
                for inner_key, inner_val in inner_dic.items():
                    res_dic[key].append({key + sep + inner_key:inner_val})
        else:
            res_dic[key] = val
    return res_dic
```
and then change here to:
```python
for dics in flatten(dic['display_fields'], nest_keys=['labels'])['labels']:
    display_fields.append(DisplayField(field=list(dics.keys())[0], display=list(dics.values())[0]))
```

Reply:
Among all of the solutions, I personally prefer the current one for the most.

Reply:
I like your third suggestion, since it would allow us to set a default order for the columns.